tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -13236,6 +13205,164 @@ void ExecutionTest::WaveSizeTest() {
+          MAX_WAVESIZE) "*2,1,1)]\r\n"
+                        "void main(uint3 tid : "
+                        "SV_DispatchThreadID ) { \r\n"
+                        "  data[tid.x].count = "
+                        "WaveActiveSum(1); \r\n" // TODO:
+                                                 // Use
+                                                 // WaveGetLaneCount()
+                                                 // instead
+                        "}\r\n";
+
+  struct WaveSizeTestData {
+    uint32_t count;
+  };
+
+  for (UINT minShaderWaveSize = 4; minShaderWaveSize <= maxWaveSize;
+       minShaderWaveSize *= 2) {
+    for (UINT maxShaderWaveSize = minShaderWaveSize * 2;
+         maxShaderWaveSize <= 128; minShaderWaveSize *= 2) {
+      for (UINT prefShaderWaveSize = minShaderWaveSize;
+           prefShaderWaveSize <= maxShaderWaveSize; prefShaderWaveSize *= 2) {
+        // format compiler args
+        char compilerOptions[64];
+        VERIFY_IS_TRUE(
+            sprintf_s(compilerOptions, sizeof(compilerOptions),
+                      "-D MINWAVESIZE=%d -D MAXWAVESIZE=%d -D PREFWAVESIZE=%d",
+                      minShaderWaveSize, maxShaderWaveSize,
+                      prefShaderWaveSize) != -1);
+
+        // run the shader
+        std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTestAfterParse(
+            pDevice, m_support, "WaveSizeTest",
+            [&](LPCSTR Name, std::vector<BYTE> &Data, st::ShaderOp *pShaderOp) {
+              VERIFY_IS_TRUE((0 == strncmp(Name, "UAVBuffer0", 10)));
+              pShaderOp->Shaders.at(0).Arguments = compilerOptions;
+              pShaderOp->Shaders.at(0).Text = waveSizeTestShader;
+
+              VERIFY_IS_TRUE(sizeof(WaveSizeTestData) * MAX_WAVESIZE <=
+                             Data.size());
+              WaveSizeTestData *pInData = (WaveSizeTestData *)Data.data();
+              memset(pInData, 0, sizeof(WaveSizeTestData) * MAX_WAVESIZE);
+            },
+            ShaderOpSet);
+
+        // verify expected values
+        MappedData dataUav;
+        WaveSizeTestData *pOutData;
+
+        LogCommentFmt(
+            L"Verifying test result for these shader wave size ranges:");
+        LogCommentFmt(L"Minimum shader wave size: %d", minShaderWaveSize);
+        LogCommentFmt(L"Maximum shader wave size: %d", maxShaderWaveSize);
+        LogCommentFmt(L"Preferred shader wave size: %d", prefShaderWaveSize);
+
+        bool AcceptedByRuntime = TestShaderRangeAgainstRequirements(
+            minShaderWaveSize, maxShaderWaveSize, minWaveSize, maxWaveSize);
+        if (AcceptedByRuntime) {
+
+          test->Test->GetReadBackData("UAVBuffer0", &dataUav);
+          VERIFY_ARE_EQUAL(sizeof(WaveSizeTestData) * MAX_WAVESIZE,
+                           dataUav.size());
+          pOutData = (WaveSizeTestData *)dataUav.data();
+
+          for (unsigned i = 0; i < MAX_WAVESIZE; i++) {
+            if (!VERIFY_IS_TRUE(pOutData[i].count >= minWaveSize &&
```

Comment:
I don't think `VERIFY_IS_TRUE` needs the `if` and `break` (although the original WaveSize test is written that way). If the verify condition is false the test fails and stops executing any further.

Reply:
Ok, I've removed the if condition for the wave size range test.